src/state-mutation.js

```diff
@@ -121,12 +121,16 @@ export const updateOrInsertResource = (state, resource) => {
     const idx = resources.findIndex(item => item.id === resource.id);
 
     const relationships = {};
-    for (const relationship in resource.relationships) {
-      if (!resource.relationships[relationship].data) {
+    for (const relationship in resources[idx].relationships) {
```

Comment:
This change seems weird to me. Maybe I just cannot comprehend what's happening, but by iterating on existing relationships, don't we neglect the relationships from the response?

Reply:
Well no, because just below we have these lines : 
```
    if (!resource.hasOwnProperty('relationships')) {
      Object.assign(resource, { relationships });
    } else {
      Object.assign(resource.relationships, relationships);
    }
```
So what happens is, we keep all the relationships available in the response as is (if it's here but empty it will remain empty), and we add the relationships we already had before, that are not present in the response.

The key line is `if (!hasOwnProperties(resource, ['relationships', relationship, 'data'])) {` because we don't update the reponse if the relationship is there.